devel/addon/ingressnginx/install.sh

```diff
@@ -84,5 +87,6 @@ helm upgrade \
     --set controller.config.no-tls-redirect-locations="" \
     --set admissionWebhooks.enabled=false \
     --set controller.admissionWebhooks.enabled=false \
+    --set controller.watchIngressWithoutClass="${INGRESS_WITHOUT_CLASS}" \
```

Comment:
This is because this PR bumps Ingress-NGINX controller's version to `v1.0.2` when running on kube 1.22. `v1.X` Ingress-NGINX has some breaking changes ([migration guide](https://kubernetes.github.io/ingress-nginx/#faq-migration-to-apiversion-networkingk8siov1)), including by default requiring the ingress class name to be defined on `Ingress` resources.
`cert-manager` ingress-shim creates `Ingress`es with our without class name depending on how the users have configured the HTTP-01 solver on the issuer, see [the code](https://github.com/jetstack/cert-manager/blob/master/pkg/apis/acme/v1/types_issuer.go#L218-L247). Our e2e tests create some `Ingress`es without the class name (and we probably want to keep doing that, at least for as long as we support Ingress-NGINX versions where the class name is not mandatory) so the tests break with `v1` Ingress-NGINX as the `Ingress`es without class name are ignored by the controller. Adding this flag makes the controller watch `Ingress`es without class.

I think we should eventually test with a standard `v1` ingress controller configuration (that only wathces `Ingress`es with class name), but I think we don't want to deploy two ingress controllers, so might need to allow for different test configurations for `v1` and pre-`v1` ingress controllers.

Reply:
Ok. I think I get it. Thanks for the explanatory comment. 
It might be worth summarizing that explanation as a comment in the code, for the benefit of others who might be tempted to change the versions etc.